### PR TITLE
gio: implement FromIterator for ListStore

### PR DESCRIPTION
**Pull request by pbor**
Fixes https://github.com/gtk-rs/gtk-rs-core/issues/951

### CODE REVIEW

**Comment by pbor** [comment]
Mind you, I have no idea what I am doing :)

File: gio/src/list_store.rs [hunk 1]
```diff
@@ -107,6 +107,16 @@ impl ListStore {
     }
 }
 
+impl<P: IsA<glib::Object>> std::iter::FromIterator<P> for ListStore {
```
Comment (sdroege) [inline gio/src/list_store.rs:110]: Could also implement it in addition for `&P` but looks good otherwise!

Comment (sdroege) [reply gio/src/list_store.rs:110]: Apparently not possible, causes conflicting impls

Comment (pbor) [reply gio/src/list_store.rs:110]: That does not seem to work: 

```
110 | impl<P: IsA<glib::Object>> std::iter::FromIterator<P> for ListStore {
    | ------------------------------------------------------------------- first implementation here
...
120 | impl<P: IsA<glib::Object>> std::iter::FromIterator<&P> for ListStore {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `auto::list_store::ListStore`
    |
    = note: downstream crates may implement trait `glib::IsA<glib::Object>` for type `&_`
```

**Review by sdroege — APPROVED** [review]

**Pull request merged**